[{"strategy": "retry", "code": "\npackage com.binance.connector.client.utils;\n\nimport org.json.JSONException;\nimport org.json.JSONObject;\nimport org.junit.Test;\nimport static org.junit.Assert.*;\n\npublic class JSONParser_buildJSONStringTest {\n\n    @Test\n    public void testBuildJSONString_Success() {\n        // Given\n        Object id = 123;\n        String method = \"testMethod\";\n        JSONObject parameters = new JSONObject();\n        parameters.put(\"param1\", \"value1\");\n\n        // When\n        String result = JSONParser.buildJSONString(id, method, parameters);\n\n        // Then\n        assertEquals(\"{\\\"method\\\":\\\"testMethod\\\",\\\"id\\\":123,\\\"params\\\":{\\\"param1\\\":\\\"value1\\\"}}\", result);\n    }\n\n    @Test(expected = JSONException.class)\n    public void testBuildJSONString_JSONException() {\n        // Given\n        Object id = new Object() {\n            @Override\n            public String toString() {\n                throw new RuntimeException(\"Simulated exception\");\n            }\n        };\n        String method = \"testMethod\";\n        JSONObject parameters = new JSONObject();\n\n        // When\n        JSONParser.buildJSONString(id, method, parameters);\n    }\n}\n"}]